crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Generic.hs

```diff
@@ -1339,7 +1398,15 @@ mergeMem c x y =
 -- to the solver and get (overapproximate) concrete answers.
 
 data SomeAlloc sym =
-  forall w. SomeAlloc AllocType Natural (Maybe (SymBV sym w)) Mutability Alignment String
+  forall w. IsSymInterface sym => SomeAlloc AllocType Natural (Maybe (SymBV sym w)) Mutability Alignment String
```

Comment:
If it works to do so, I'd rather see this constraint propagated to the `Eq` instance below.  It's usually better to avoid packaging up typeclass dictionaries into existential types unless it's really necessary.